src/test/java/com/hubspot/jinjava/lib/tag/eager/EagerImportTagTest.java

```diff
@@ -436,6 +436,15 @@ public void itDefersQuadLayer() {
     assertThat(interpreter.render(result).trim()).isEqualTo("12345 cbaabaaba");
   }
 
+  @Test
+  public void itImportsDoublyNamed() {
+    setupResourceLocator();
+    String result = interpreter.render(
+      "{% import 'variables.jinja' as foo %}{{ foo.foo['foo'].bar }}"
```

Comment:
Before this change, `foo.foo` would be undefined, as we were removing the `foo` key from the child bindings to prevent a reference loop. In this case they're not referencing the same thing so it's fine to put the imported `foo` as a key within the `foo` alias.
(I know, it's a bit hard to explain which foo is which, but they need to have the same name for this bug)